compiler/noirc_frontend/src/monomorphization/tests.rs

```diff
@@ -121,23 +114,24 @@ fn simple_closure_with_no_captured_variables() {
     }
     "#;
 
-    let expected_rewrite = r#"fn main$f0() -> Field {
-    let x$0 = 1;
-    let closure$3 = {
-        let closure_variable$2 = {
-            let env$1 = (x$l0);
-            (env$l1, lambda$f1)
+    let program = get_monomorphized!(src, Expect::Success).unwrap();
+    insta::assert_snapshot!(program, @r"
```

Comment:
I've never used `insta`; a quick glance at the docs suggested that it would store some kind `.snap` file in a `snapshots` directory, and later compare against that. But here we're comparing against some hardcoded string, so I wonder if `similar_asserts::assert_eq!` we use to compare SSAs would also be an option? It provides diff-like presentation as well. Is `insta::assert_snapshot!` better in some way?

Reply:
If I just did `insta::assert_snapshot(program)` then we'd write out the value of `format!("{display}")` into a .snap file and compare against that as you said.

Here I'm using the inline snapshot feature (showm by preceding the string by an `@`), this means that whenever we update the snapshot's expected result it's automatically written into the following string literal. 

The main benefit here over just using `similar_asserts::assert_eq!` is when we intentionally modify the code in a way which would change the snapshots. Rather than needing to run each test and copy the new result from the terminal into the codebase one by one, we instead just run the tests and can review and update the snapshots by just running `cargo insta review`.